### DISCUSSION TITLE
Support for the Load Control

### DISCUSSION THREAD

**Discussion by michalpulda** [mletenay/home-assistant-goodwe-inverter#157] [Q&A] [answered]
Hello,

Would it be possible to add support for the Load Control? I have a boiler connected to the Load Control and it serves as an easy/cheap version of a wattrouter. However, I would like to create more complex automations (i.e. take in account low tariff during night).

What the Load Control is can be found in the [PV Master manual](https://kr.goodwe.com/Ftp/EN/Downloads/User%20Manual/GW_PV%20Master_User%20Manual-EN.pdf) (page 33-34, or see bellow).

Thanks!

![GW_PV Master_User Manual-EN_1](https://user-images.githubusercontent.com/5116915/217550984-83c27db6-fbe5-403d-9e1c-46857935b215.png)
![GW_PV Master_User Manual-EN_2](https://user-images.githubusercontent.com/5116915/217550993-04eb1484-80fd-4a04-9fd1-f36aec7132e7.png)

**Comment by janchlebek** [comment]
Not an actual answer from me but since we are in Home Assistant world: I think you can achieve all this by an automation outside of GoodWe and all the flexibility is already there (I am doing it that way already). Only thing you need is to have your boiler controlled by some relay or switch that is accessible to HA.

**Comment by mletenay** [comment]
I have noticed that feature in new firmwares and for a moment considered implementing it.
But then had exactly the same thought as @janchlebek above - for most people it is easier to use some independent relay like sonoff and control that from HA rather then wiring something directly into inverter.
But it may not be a big deal, I'll look into it ...

**Reply by michalpulda** [reply]
Thank you for considering looking into it and thanks @janchlebek for the answer above!

I've been thinking about using smart relay instead of the Load Control and it has pros and cons:
- pro: while using a smart relay I would be able to create whatever automation I can think of
- con: before I had my PV system installed, I've already tried 2 or 3 plugs/smart relays (sonoff, Shelly plug and Shelly 1 PM) to automate the boiler, but even though these devices are rated up to 16 A, these are not comfortable taking high load over long time. These devices will get quite hot (in a sense of "I can smell melting plastics" hot) and I don't want to risk any fire hazard.
- con: these devices have limited life span and I have to deal with family acceptance factor. Having cold water is not acceptable. I have, say, 20 Shelly devices all around the flat, and just recently one malfunctioned (after working flawlessly for a year and a half) and yes, my family protested. It was behind venetian blinds switch, so imagine the fuss if it was controlling hot water. :-)
- yes, I didn't use Shelly Pro 1 PM (or any good alternative), so if anyone can vouch for it, I will take it!

Anyhow, I would like to stay in the HA world. Yet I would like to stay at the safe site of the family acceptance factor as well. I can imagine if the Load Control would be accessible from the HA I would have both. HA's automations and reliability of the invertor in combination with a relay as [this one](https://datasheet.eaton.com/datasheet.php?model=265574) should make it.

Thanks!

**Reply by janchlebek** [reply]
Hi, I totally understand your concerns - I was few times facing family acceptance issues with various automations as well - here I can only suggest to think all them well :)
Also I do understand your load issues with smart plugs - for this one I can suggest either using a shelly or similar switch to control a traditional contactor rated for your load or use a robust smart switch. I personally switched my boiler to Sonoff POWR3 device which is rated for 25A and works really well. Also it is a normally closed type of relay so if the device would fail at some point the load remains  on (i.e. hot water always works). On top it can easily be flashed with Tasmota to remain local.

**Reply by michalpulda** [reply]
Thank you @janchlebek for the tip! If the functionality won't be implemented, I can go that route. Thanks!

**Comment by davefx** [comment]
No matter of if this is the best option or not: If the inverter already has
a controllable relay, IMHO it's a good idea to allow its control from HA...
Controllable relays are not free :-)
--
David Marín Carreño ***@***.***>

**Reply by mletenay** [reply]
@michalpulda Do you actually see the Load Control items in PvMaster ?
Since I have the ET with latest firmware and I cannot see any such thing there.

**Reply by janchlebek** [reply]
> No matter of if this is the best option or not: If the inverter already has a controllable relay, IMHO it's a good idea to allow its control from HA... Controllable relays are not free :-) -- David Marín Carreño ***@***.***> El jue, 9 feb 2023 a las 7:54, janchlebek ***@***.***>) escribió:
> […](#)

Yes - agree, makes complete sense to have this in the integration, however I personally think an independent relay or switch controlled via HA gives the user WAY MORE control over the load connected via HA automations than the inverter itself offers. That easily justifies some €40-€50 for the relay.

**Reply by michalpulda** [reply]
@mletenay Yes, I can see it as shown on the printscreen bellow. My inverter (GW10K-ET Plus) is on firmware version 080822. As far as I know (= what my installation company told me) only Plus version of the inverter has a dry contact used for the load control. If there is anything I can help with, please let me know. Thanks!

![Screenshot_20230210_174545_PV Master](https://user-images.githubusercontent.com/5116915/218148636-10034e6a-e811-445c-bdd3-a747a18f79ad.jpg)

**Reply by michalpulda** [reply]
> Yes - agree, makes complete sense to have this in the integration, however I personally think an independent relay or switch controlled via HA gives the user WAY MORE control over the load connected via HA automations than the inverter itself offers. That easily justifies some €40-€50 for the relay.

Having this functionality exposed to HA would give you the same amount of control, no?

**Reply by mletenay** [reply]
OK, I see now. It is indeed present only on ET+ inverter, the ET does not have that DO relay present at all.
That means it won't be so easy, since I can't test it locally and would need someone to test it ...
If you are willing to do it, than we should move the discussion to discord channel and continue there.

**Reply by janchlebek** [reply]
> > Yes - agree, makes complete sense to have this in the integration, however I personally think an independent relay or switch controlled via HA gives the user WAY MORE control over the load connected via HA automations than the inverter itself offers. That easily justifies some €40-€50 for the relay.
> 
> Having this functionality exposed to HA would give you the same amount of control, no?

Yes and no - you need to do the automation anyway of course but then you can either control the load directly via some stupid relay while being independent from relatively complicated path via the inverter. I think it is easier and a little more reliable/failproof to go via relay as that cannot be influenced by communication with inverter and also (depending on the product  used) gives you actual consumption readings from the relay (to see if the load is actually working or not for example), etc.
This is just my opinion, like with almost everything in home automation you have multiple ways to accomplish more or less the same but it is not always the same :)

**Reply by michalpulda** [reply]
@mletenay Of course I am willing to test. I sent you a DM on Discord. Thanks!
@janchlebek Fair points! I can imagine that ethernet connected Shelly Pro can be more stable than a wifi connection (even though I have an AP like 150 cm far from the inverter). Also to see that the boiler is actually working from consumption readings has some merit. Thanks!

**Comment by mletenay** [comment, marked as answer]
Load control switch was added to v0.9.9.7.

**Reply by michalpulda** [reply]
Thanks! I have tried it and it doesn't work. How can I help with troubleshooting?

**Reply by mletenay** [reply]
Should be fixed now.

**Reply by michalpulda** [reply]
Yes, I can confirm that I turning the load control from Home Assistant changes the status in PV Master. Turning the load control on and off doesn't change the status in HA, but it is not a problem for me. Thanks again.
